src/app/forms/custom-sections/letters/letters.component.ts

```diff
@@ -73,11 +76,22 @@ export class LettersComponent implements OnInit, OnDestroy, OnChanges {
       }
       : undefined;
   }
+  // checking if the Technical Record History has current status code.
+  get checkRecordHistoryHasCurrent(): boolean {
+    let ifCurrent = false;
+    this.techRecordService.techRecordHistory$.subscribe((historyArray: TechRecordSearchSchema[] | undefined) => {
+      historyArray?.forEach((history: TechRecordSearchSchema) => {
+        if (history.techRecord_statusCode === StatusCodes.CURRENT
+          && this.techRecord?.techRecord_statusCode === StatusCodes.PROVISIONAL) ifCurrent = true;
+      });
+    });
+    return ifCurrent;
+  }
 
   get eligibleForLetter(): boolean {
-    const currentTechRecord = this.techRecord?.techRecord_statusCode === StatusCodes.CURRENT;
-
-    return this.correctApprovalType && currentTechRecord && !this.isEditing;
+    const currentTechRecord = this.techRecord?.techRecord_statusCode === StatusCodes.CURRENT
```

Comment:
This shouldn't be called `currentTechRecord` if it could be provisional too

Reply:
Yes, I will correct this line and give it a proper name.